data/src/main/java/com/microsoft/azure/kusto/data/auth/ManagedIdentityTokenProvider.java

```diff
@@ -0,0 +1,37 @@
+package com.microsoft.azure.kusto.data.auth;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.identity.ManagedIdentityCredential;
+import com.azure.identity.ManagedIdentityCredentialBuilder;
+import com.microsoft.azure.kusto.data.exceptions.DataServiceException;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.net.URISyntaxException;
+
+public class ManagedIdentityTokenProvider extends TokenProviderBase {
+    private final ManagedIdentityCredential managedIdentityCredential;
+    private final TokenRequestContext tokenRequestContext;
+
+    public ManagedIdentityTokenProvider(String managedIdentityClientId, @NotNull String clusterUrl) throws URISyntaxException {
+        super(clusterUrl);
+        if (StringUtils.isNotBlank(managedIdentityClientId)) {
+            this.managedIdentityCredential = new ManagedIdentityCredentialBuilder()
+                    .clientId(managedIdentityClientId) // only required for user assigned
+                    .build();
+        } else {
+            this.managedIdentityCredential = new ManagedIdentityCredentialBuilder().build();
+        }

```

Comment:
```suggestion
       ManagedIdentityCredentialBuilder builder = new ManagedIdentityCredentialBuilder();
        if (StringUtils.isNotBlank(managedIdentityClientId)) {
            builder = builder.clientId(managedIdentityClientId); // only required for user assigned
        }
        this.managedIdentityCredential = builder.build();
```
Cutting down on a bit of duplication